src/index.js

```diff
@@ -50,6 +51,14 @@ const isReactComponent = (value: any) =>
   value && value.$$typeof === reactElement;
 
 function diffStrings(valueA: any, valueB: any, options: Options) {
+  // jest-diff doesn't support single-line values, but we want to
```

Comment:
this seems weird, IMO. Maybe not, though :D

Reply:
Maybe that's a regression? 
https://github.com/facebook/jest/issues/1633
cc @pedrottimark

Reply:
Hah, there's even [a test](https://github.com/facebook/jest/blob/03bf2a93c7b99d08ccad1a86ace4c99555f83017/packages/jest-diff/src/__tests__/diff.test.js#L84-L91) for that.

Is this because `jest-diff` was supposed to provide an inline diff, but it was not supported at the time (and nothing change in that matter for now)?

Reply:
Y’all ask a good question about this special case. I added test to document it, not endorse it :)

I guess it’s because `jest-diff` omits output that seems redundant when `expect` calls it:

<img width="300" alt="2018-01-31 single-line" src="https://user-images.githubusercontent.com/11862657/35629072-b27e8dea-066b-11e8-9853-646dacc40f46.png">

but it’s confusing when `snapshot-diff` calls it.

Not to sound critical of awesome effort to integrate expect matchers with diff display, and then snapshots, it feels like the whole is less than the sum of the two parts in some cases. What do you think if expect omits `Expected value to equal` and `Received` lines when a diff follows?

Yesterday I updated a local copy of code from dormant Jest PR for inline substring diff to call `diff-sequences` package. This version is somewhat easier to grok than the original :)

Any feedback how to expose that new function for Jest and other callers is very welcome.

Reply:
@pedrottimark you could expose some useful utils like `diff` lib (talking about something like `diffLines` and `structuredPatch` we use).
Back to the topic, `diff` + `expect` behavior feels like a hack to me. It's `.toEquals()` matcher that should be responsible for determining whether to display the diff or not.
Do you think it makes sense to file a PR to Jest?

Reply:
Yeah, we might refactor to more clearly separate assertion logic at one extreme and diff display at another extreme from the responsibility you mention, which I think is divided between `expect/src/matchers.js` and `jest-diff/src/index.js` files.